### DISCUSSION TITLE
Ghost menu's

### DISCUSSION THREAD

**Discussion by PH5HP** [e107inc/e107#5302] [General]
Hi,

I wonder what is happening. I deleted some menu's and they still appear in the menu list. First screenshot is the menu list in the page.menu section.

![image](https://github.com/user-attachments/assets/e4605127-ac7d-41f3-a3f8-37b7f54e36a4)

Second screenshot is the menu manager.

![image](https://github.com/user-attachments/assets/83fe9f0b-58a7-4369-98cf-c920612e2028)

There are some menu's in the menu manager that are not present in the page/menu section. Anyone? Bug? I tried to clear the Chrome browser cache and refresh the page, but no luck.

Kind regards,
Henk - ph5hp.nl

**Comment by Jimmi08** [comment]
I suggest reading this from the end:  Shortly: menus in Page/Menus are not custom menus in Menu Manager.   

https://github.com/e107inc/e107/issues/3831

**Reply by PH5HP** [reply]
Hi Jimmi08,

I did read #3831. Thank you.

But I still do not understand what is happening. You told me that 'Shortly: menus in Page/Menus are not custom menus in Menu Manager.'

![image](https://github.com/user-attachments/assets/a4321cf0-e256-4ac5-86ae-a99ea65634d7)

But the 'clocklink' menu is a custom menu (See screenshot above). It is listed in the 'custom menu tab'. 

The reason for this thread is simple. I have some menus in the tab of the custom menus in the menu manager which are already deleted in the page/menu section. I checked and they do not exists anymore there. I understand it was an issue some years ago, but they solved this? Also is '# 18' not a menu but a page. Strange. It appears also in the menu manager custom menus tab. '# 19 ' and '# 21' are also deleted already.

I am not an expert in E107, but I gave it a try again after some difficult years of using WP. E107 v0.xx and V1 was my first CMS and I enjoyed using it. So I went back to E107.

Kind regards,
Henk - ph5hp.nl

**Reply by Jimmi08** [reply]
`Also is '# 18' not a menu but a page.`   - check it, click on tab Menu, check filled fields (if image, title or name on menu tab is filled, it is in menu manager) 
At that issue, I asked for the same rule for menu definition (f.e. menu name filled).   It was declined, so if you fill some value in menu tab - mostly image - it is listed in menu manager.

The menu and page are the same table record. I recommend using only the Overview section to see all records in this table.  Check there iD 18 and tab Menu...  I am sure that something is filled. 

Check if those records are deleted from the database (via phpmyadmin)  for real - I had an issue above this too some years ago.

**Reply by PH5HP** [reply]
Hi,

I checked it and nothing was filled with data in the menu tab. Strange is that '# 18' has disappeared in the menu manager custom menus, but there is now a '# 10'. I checked '# 10' and that is also a page. I changed nothing at all. So there is clearly something wrong (see screenshot below).

![image](https://github.com/user-attachments/assets/bdaf4775-ff25-41b7-bb31-9abc24683034)

The menus '# 19' and '# 21' are not present in the E107_page table of the E107 database. The menus '# 10' and '# 18' are present in the table. They are not menus, but in fact pages. I checked them too to be sure the menu fields are empty.

What can I check more?

Kind regards,
Henk - ph5hp.nl

**Reply by PH5HP** [reply]
Hi,

I found out that they are still present in the e107_menus table. Can I delete them?

Kind regards,
henk - ph5hp.nl

**Reply by Jimmi08** [reply]
Table e107_menus , Oh and clear cache. 

![image](https://github.com/user-attachments/assets/1913cf96-c0e2-4334-9c94-0cf4caa034c0)

If menu_location is  0,  it is not used, you can delete it.  It will be added automatically when going to menu manager.

**Reply by PH5HP** [reply]
Hi,

Thank you. Can you tell me why menus appear more than one time in that table? Even if they are not used? Would it not be better to set the menu_location to '0' and keep just one in the table if it is not used at all?

Kind regards,
Henk - ph5hp.nl

**Reply by PH5HP** [reply]
Never mind :-) .. I got it ... To be able to select it from the list.
